pysages/ssages/methods/umbrella.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2020-2021: PySAGES contributors
+# See LICENSE.md and CONTRIBUTORS.md at https://github.com/SSAGESLabs/PySAGES
+
+from typing import NamedTuple
+from .core import SamplingMethod, generalize  # pylint: disable=relative-beyond-top-level
+import jax.numpy as np
+from jaxlib.xla_extension import DeviceArray as JaxArray
+
+
+class UmbrellaState(NamedTuple):
+    """
+    Description of an umbrella sampling state.
+
+    bias -- additional forces for the class.
+    xi -- current cv value.
+    """
+    bias: JaxArray
+    xi: JaxArray
+
+    def __repr__(self):
+        return repr("PySAGES" + type(self).__name__)
+
+
+class UmbrellaSampling(SamplingMethod):
+    def __init__(self, cvs, kspring, center, *args, **kwargs):
+        super().__init__(cvs, args, kwargs)
+        self.kspring = np.asarray(kspring)
+        self.center = np.asarray(center)
+
+    def __call__(self, snapshot, helpers):
+        return _umbrella(snapshot, self.cv, self.center, self.kspring, helpers)
+
+
+def _umbrella(snapshot, cv, center, kspring, helpers):
```

Comment:
```suggestion
    def __call__(self, snapshot, helpers):
        return _umbrella(self, snapshot, helpers)


def _umbrella(method, snapshot, helpers):
    cv = method.cv
    center = method.center.reshape(1, -1)
    kspring = self.kspring.reshape(1, -1)
```

Does this (along with the previous matmul change) fix the issue in higher dims than one?

Reply:
Ok, I played around with this and actually used my brain.
Now it works, but there is probably a faster way to write that down.
We need to sum over the different CVs.